### PR TITLE
command pr, show by number, add reviewers

### PR DESCRIPTION
**Pull request by dbaltas**
* Add reviewers on create pr
* ergo pr [number], display infor for pr

### CODE REVIEW

**Review by mantzas — COMMENTED** [review]
a usage in readme.md would be nice

File: cmd/deploy.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func deployBranches(branchMap map[string]string, suffixFind, suffixReplace strin
 	green.Printf("Deployment will start in %s\n", d.String())
 	time.Sleep(d)
 
-	for i, branch := range branches {
+	for i, branch := range releaseBranches {
```
Comment (nstratos) [inline cmd/deploy.go:107]: At the bottom of this function (line 149) new assignment of `release` is not used. Either ignore or use in `green.Printf`.

File: cmd/pr.go [hunk 2]
```diff
@@ -13,20 +14,54 @@ import (
...
 	RunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+		// show a PR by number
+		if len(args) > 0 {
+			number, err = strconv.Atoi(args[0])
+			if err != nil {
+				return err
+			}
+			err := getPR(number)
+
```
Comment (nstratos) [inline cmd/pr.go]: Remove empty line. Group error handling together e.g. 

```
err := getPR(number)
if err != nil {
	return err
}
```

Or use shorthand

```
if err := getPR(number); err != nil {
	return err
}
```

When using shorthand, be careful with variable shadowing. Check using:

    go vet --shadow

File: cmd/pr.go [hunk 3]
```diff
@@ -13,20 +14,54 @@ import (
...
+	Long: `Create a pull request on github from compare branch to base branch
+	ergo pr --title "my new pull request --reviewers pespantelis,nstratos,mantzas"
+Show details of a pr by pr number
+	ergo pr 18
+Add reviewers to an existing pr
+	ergo pr 18 --reviewers pespantelis,nstratos,mantzas"
+	`,
+	Args: cobra.MaximumNArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
+		var err error
```
Comment (nstratos) [inline cmd/pr.go]: Not sure declaring error is necessary here. Assigning with `:=` and returning the error right away might be clearer (one less thing to remember).

File: github/github.go [hunk 4]
```diff
@@ -88,15 +88,40 @@ func (gc *Client) CreatePR(baseBranch, compareBranch, title, body string) (*gith
 	return pr, nil
 }
 
+// GetPR gets a pull request
+func (gc *Client) GetPR(number int) (*github.PullRequest, error) {
+	pr, _, err := gc.client.PullRequests.Get(gc.ctx, gc.organization, gc.repo, number)
+
```
Comment (nstratos) [inline github/github.go]: Remove empty line so error handling is right after the statement.

File: github/github.go [hunk 5]
```diff
@@ -88,15 +88,40 @@ func (gc *Client) CreatePR(baseBranch, compareBranch, title, body string) (*gith
...
+// RequestReviewersForPR assigns reviewers to a pull request
+func (gc *Client) RequestReviewersForPR(number int, reviewers, teamReviewers string) (*github.PullRequest, error) {
+	payload := github.ReviewersRequest{
+		Reviewers:     strings.Split(reviewers, ","),
+		TeamReviewers: strings.Split(teamReviewers, ","),
+	}
+	fmt.Println(github.Stringify(payload))
+	pr, _, err := gc.client.PullRequests.RequestReviewers(gc.ctx, gc.organization, gc.repo, number, payload)
+
```
Comment (nstratos) [inline github/github.go]: Here as well.

File: cmd/pr.go [hunk 6]
```diff
@@ -66,5 +101,48 @@ func createPR() error {
...
+	return nil
+}
+
+func getPR(number int) error {
+	yellow := color.New(color.FgYellow)
+	green := color.New(color.FgGreen)
+	red := color.New(color.FgRed)
+
+	prp, err := gc.GetPR(number)
+	pr := *prp
```
Comment (nstratos) [inline cmd/pr.go]: When `gc.GetPR` returns an error, prp is nil so maybe move `pr := *prp` after error checking.

**Review by nstratos — APPROVED** [review]

**Pull request merged**